### PR TITLE
Add connectome-tools 0.3.4 and 0.4.0

### PR DESCRIPTION
**Pull request by GianlucaFicarelli**
* 0.4.0 depends on bluepy >= 2.0
* 0.3.4 depends on bluepy < 2.0, and it can be removed later if there are no issues with 0.4.0

### CODE REVIEW

File: deploy/environments/applications.yaml [hunk 1]
```diff
@@ -22,7 +22,8 @@ spack:
     - brion@3.3.0 +python
     - brion@3.3.1 +python
     - circuit-build
-    - connectome-tools@0.3.3
+    - connectome-tools@0.3.4
+    - connectome-tools@0.4.0
```
Comment (matz-e) [inline deploy/environments/applications.yaml:26]: Once we're deploying *only* the most current version, that `@version` should disappear.

File: var/spack/repos/builtin/packages/connectome-tools/package.py [hunk 2]
```diff
@@ -13,7 +13,8 @@ class ConnectomeTools(PythonPackage):
     git      = "ssh://bbpcode.epfl.ch/nse/connectome-tools"
 
     version('develop', branch='master')
-    version('0.3.3', tag='connectome-tools-v0.3.3', preferred=True)
+    version('0.4.0', tag='connectome-tools-v0.4.0', preferred=True)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/connectome-tools/package.py:16]: `preferred=True` is only needed if an older version should be given preference.

**Review by matz-e — APPROVED** [review]
LGTM, comments can be addressed in the future.

**Comment by GianlucaFicarelli** [comment]
Thanks, I opened NSETM-1340 to keep track of the changes to be done

**Pull request merged**